modules/gdscript/gdscript_parser.cpp

```diff
@@ -5410,17 +5423,21 @@ void GDScriptParser::TreePrinter::print_while(WhileNode *p_while) {
 }
 
 void GDScriptParser::TreePrinter::print_tree(const GDScriptParser &p_parser) {
-	ERR_FAIL_COND_MSG(p_parser.get_tree() == nullptr, "Parse the code before printing the parse tree.");
+	ClassNode *class_tree = p_parser.get_tree();
+	ERR_FAIL_NULL_MSG(class_tree, "Parse the code before printing the parse tree.");
 
 	if (p_parser.is_tool()) {
 		push_line("@tool");
 	}
-	if (!p_parser.get_tree()->icon_path.is_empty()) {
+	if (class_tree->is_abstract) {
+		push_line("@abstract");
+	}
```

Comment:
This works because [`test_parser()`](https://github.com/godotengine/godot/blob/6916349697a4339216469e9bf5899b983d78db07/modules/gdscript/tests/test_gdscript.cpp#L121-L122) also uses analyzer. If we change this, a better solution would be to check class annotations that have already been added in the parser, but have not yet been applied (excluding `@icon`, which is used in the parser without adding to the AST). No need to do anything, just a note.